### PR TITLE
Add nightly build

### PR DESCRIPTION
**Pull request by jsturtevant**
Signed-off-by: James Sturtevant <jstur@microsoft.com>

**Description**

Allows for using changes from the main branch in testing.

I got this working on my fork: https://github.com/jsturtevant/azure-capi-cli-extension/releases/tag/az-capi-nightly

**History Notes**

---

This checklist is used to make sure that common guidelines for an Azure CLI pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: az-capi-nightly
+
+on:
+  push:
+    branches: [ master ]
```
Comment (mboersma) [inline .github/workflows/nightly.yml:5]: Shouldn't this be `[ main ]`?

**Review by mboersma — COMMENTED** [review]
This is great, thanks @jsturtevant!

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (jsturtevant) [reply .github/workflows/nightly.yml]: fixed

**Comment by jsturtevant** [comment]
woops messed up rebase... will fix shortly

**Comment by jsturtevant** [comment]
@mboersma should be all set

**Review by mboersma — APPROVED** [review]
lgtm

**Pull request merged**